Helm V3 alpha 2 Segmentation fault

Hello,

Been using alpha 1 for some time, tried V2 with our standard workflow (we were already using the --wait flag so we wanted to see the difference with alpha2).

Sadly first deployment was installing and then panic : 

```
Deploying new release...
Release "RELEASE-NAME" does not exist. Installing it now.
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x127b21b]

goroutine 1 [running]:
helm.sh/helm/pkg/kube.(*waiter).statefulSetReady(0xc003628100, 0xc000886000, 0x18)
	/go/src/helm.sh/helm/pkg/kube/wait.go:270 +0x6b
helm.sh/helm/pkg/kube.(*waiter).waitForResources.func1(0xc000a838a8, 0xc000a83890, 0x2)
	/go/src/helm.sh/helm/pkg/kube/wait.go:146 +0x8fd
helm.sh/helm/vendor/k8s.io/apimachinery/pkg/util/wait.WaitFor(0xc000bd2000, 0xc001a384e0, 0xc0012d05a0, 0x0, 0x0)
	/go/src/helm.sh/helm/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:426 +0x137
helm.sh/helm/vendor/k8s.io/apimachinery/pkg/util/wait.pollInternal(0xc000bd2000, 0xc001a384e0, 0x0, 0x0)
	/go/src/helm.sh/helm/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:312 +0x8a
helm.sh/helm/vendor/k8s.io/apimachinery/pkg/util/wait.Poll(0x77359400, 0x45d964b800, 0xc001a384e0, 0x2, 0x2)
	/go/src/helm.sh/helm/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:306 +0x4d
helm.sh/helm/pkg/kube.(*waiter).waitForResources(0xc003628100, 0xc0004d0400, 0x29, 0x40, 0x29, 0x40)
	/go/src/helm.sh/helm/pkg/kube/wait.go:50 +0x14b
helm.sh/helm/pkg/kube.(*Client).Wait(0xc0001b2960, 0x1b0f9c0, 0xc001487920, 0x45d964b800, 0xc694, 0xe000)
	/go/src/helm.sh/helm/pkg/kube/client.go:101 +0x11c
helm.sh/helm/pkg/action.(*Install).Run(0xc0007392c0, 0xc000177a00, 0x5, 0x6, 0x0)
	/go/src/helm.sh/helm/pkg/action/install.go:212 +0x931
main.runInstall(0xc00044aa80, 0x2, 0x32, 0xc0007392c0, 0x1b127c0, 0xc00000e018, 0x1, 0x35, 0x0)
	/go/src/helm.sh/helm/cmd/helm/install.go:213 +0x36f
main.newUpgradeCmd.func1(0xc0000d6280, 0xc00044aa80, 0x2, 0x32, 0x0, 0x0)
	/go/src/helm.sh/helm/cmd/helm/upgrade.go:101 +0x728
helm.sh/helm/vendor/github.com/spf13/cobra.(*Command).execute(0xc0000d6280, 0xc00044a380, 0x32, 0x38, 0xc0000d6280, 0xc00044a380)
	/go/src/helm.sh/helm/vendor/github.com/spf13/cobra/command.go:762 +0x465
helm.sh/helm/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc000528f00, 0x1b12700, 0xc00000e018, 0xc00003ae10)
	/go/src/helm.sh/helm/vendor/github.com/spf13/cobra/command.go:850 +0x2fc
helm.sh/helm/vendor/github.com/spf13/cobra.(*Command).Execute(...)
	/go/src/helm.sh/helm/vendor/github.com/spf13/cobra/command.go:800
main.main()
	/go/src/helm.sh/helm/cmd/helm/helm.go:75 +0xd2


```

What information would you need to help you debug that ?

--- a/pkg/kube/wait.go
+++ b/pkg/kube/wait.go
@@ -267,7 +267,10 @@ func (w *waiter) statefulSetReady(sts *appsv1.StatefulSet) bool {
 	var partition int
 	// 1 is the default for replicas if not set
 	var replicas = 1
-	if sts.Spec.UpdateStrategy.RollingUpdate.Partition != nil {
+	// For some reason, even if the update strategy is a rolling update, the
+	// actual rollingUpdate field can be nil. If it is, we can safely assume
+	// there is no partition value
+	if sts.Spec.UpdateStrategy.RollingUpdate != nil && sts.Spec.UpdateStrategy.RollingUpdate.Partition != nil {
 		partition = int(*sts.Spec.UpdateStrategy.RollingUpdate.Partition)
 	}
 	if sts.Spec.Replicas != nil {

